Cabal/Distribution/Simple/LocalBuildInfo.hs

```diff
@@ -152,6 +161,37 @@ data LocalBuildInfo = LocalBuildInfo {
 
 instance Binary LocalBuildInfo
 
+-- | Describes what components are enabled by user-interaction.
+data ComponentEnabledSpec
+    = ComponentEnabledSpec {
+        testsEnabled :: Bool,
+        benchmarksEnabled :: Bool
```

Comment:
will this have e.g. `executablesEnabled` flag. Can it be uses to more precise selection of buildable components (i.e. only single test-suite)? If yes, could we directly have the target `Spec`, even we don't use all features of it?